src/state/opamSysInteract.ml

```diff
@@ -1093,14 +1105,50 @@ let install_packages_commands_t ?(env=OpamVariable.Map.empty) config sys_package
     [`AsUser (Commands.msys2 config),
      "-Su"::"--noconfirm"::packages], None
   | Netbsd -> [`AsAdmin "pkgin", yes ["-y"] ("install" :: packages)], None
+  | Nix ->
+    (match switch with
```

Comment:
To avoid passing switch option, is it possible to add as optional argument a function that returns the needed prefix?

Reply:
Yes, I think so

Reply:
After review i found that instead the switch state was more indicated if at least to avoid uses of the unsightly `OpamStateConfig.(!r.root_dir)`. This is implemented in the fixup commit https://github.com/ocaml/opam/pull/5982/commits/241d29cbc3594c9bee6126f29d713c32eb425c22